tests/integration/cli/commands/test_issue_1741.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import json
+import os
+
+import pytest
+
+from pex.cli.testing import run_pex3
+from pex.pep_440 import Version
+from pex.pep_503 import ProjectName
+from pex.resolve.lockfile import json_codec
+from pex.testing import PY_VER, run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+@pytest.mark.skipif(
+    PY_VER < (3, 6), reason="opentelemetry-instrumentation-httpx<0.31 requires python >= 3.6"
+)
+def test_prereleases(
+    tmpdir,  # type: Any
+):
+    # type: (...) -> None
+
+    lockfile = os.path.join(str(tmpdir), "lock")
+    run_pex3(
+        "lock",
+        "create",
+        "opentelemetry-instrumentation-httpx[instruments]<0.31",
+        "--pre",
+        "-o",
+        lockfile,
+        "--indent",
+        "2",
+    ).assert_success()
+
+    lock = json_codec.load(lockfile)
+    assert 1 == len(lock.locked_resolves)
+    locked_resolve = lock.locked_resolves[0]
+    versions_by_project_name = {
+        locked_requirement.pin.project_name: locked_requirement.pin.version
+        for locked_requirement in locked_resolve.locked_requirements
+    }
+    expected_version = versions_by_project_name[ProjectName("opentelemetry-instrumentation-httpx")]
+    assert expected_version.parsed_version.is_prerelease
+
+    use_lock_args = [
+        "--lock",
+        lockfile,
+        "--",
+        "-c",
+        "from opentelemetry.instrumentation.httpx.version import __version__; print(__version__)",
+    ]
+
+    # 1st prove this does the wrong thing on prior broken versions of Pex.
+    result = run_pex_command(
+        args=["pex==2.1.83", "-c", "pex", "--"] + use_lock_args,
+        quiet=True,
+    )
+    result.assert_failure()
```

Comment:
Very nit: I haven't followed conventions for testing in this repo, but this feels similar to leaving dead code in but commented out. You could always run this test on an older SHA to prove this (and of course it's always good to see a test fail before committing it), but leaving it in seems like overkill...?

Reply:
Not sure, I started doing this recently to prove the fix fixes instead of hacking a commented out code block locally and telling myself and reviewers to "trust me". It definitely is overkill.

Reply:
Cool. Yea, fine with me. And an easy thing to GC out for older fixes if the test suite gets too slow.

Reply:
Yeah, that's what I figure. ITs are faster / more scalable that units right now due to xdist.